sky/utils/kubernetes/rsync_helper.sh

```diff
@@ -61,7 +66,21 @@ fi
 # Timeout after MAX_WAIT_TIME_SECONDS seconds.
 MAX_WAIT_TIME_SECONDS=300
 MAX_WAIT_COUNT=$((MAX_WAIT_TIME_SECONDS * 2))
+
+# Do NOT use eval here: it will expand $1/$@ in the local shell (this script)
+# and collapse argv, breaking rsync's remote protocol invocation.
 # Use --norc --noprofile to prevent bash from sourcing startup files that might
-# output to stdout and corrupt the rsync protocol. All debug output must go to
-# stderr (>&2) to keep stdout clean for rsync communication.
-eval "${kubectl_cmd_base% --} -i -- bash --norc --noprofile -c 'count=0; until which rsync >/dev/null 2>&1; do if [ \$count -ge $MAX_WAIT_COUNT ]; then echo \"Error when trying to rsync files to kubernetes cluster. Package installation may have failed.\" >&2; exit 1; fi; sleep 0.5; count=\$((count+1)); done; exec \"\$@\"' -- \"\$@\""
+# output to stdout and corrupt the rsync protocol.
+"${kubectl_args[@]}" -i -- bash --norc --noprofile -c '
+max_count="$1"; shift
+count=0
+until command -v rsync >/dev/null 2>&1; do
+  if [ "$count" -ge "$max_count" ]; then
+    echo "Error when trying to rsync files to kubernetes cluster. Package installation may have failed." >&2
+    exit 1
+  fi
+  sleep 0.5
+  count=$((count+1))
```

Comment:
can we add a output to stderr for it is waiting for the rsync to be installed on remote?